### PR TITLE
Added functionality to create north bound task with chosen plugin [FOGL-1529]

### PR DESCRIPTION
**Pull request by monika-Dianomic**

### CODE REVIEW

File: src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+      nextContent.setAttribute('class', 'box step-content is-active');
+    }
+  }
+
+  addScheduledTask(formValues, nxtButton) {
+    /** request started */
+    this.ngProgress.start();
+    formValues['schedule_enabled'] = false,
+    formValues['schedule_repeat'] = 30,
+    formValues['cmd_params'] = { 'stream_id': '1', 'debug_level': '1' };
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts]: `{}`

File: src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+    if (nextContent != null) {
+      nextContent.setAttribute('class', 'box step-content is-active');
+    }
+  }
+
+  addScheduledTask(formValues, nxtButton) {
+    /** request started */
+    this.ngProgress.start();
+    formValues['schedule_enabled'] = false,
+    formValues['schedule_repeat'] = 30,
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts]: why?

File: src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+    const nextContent = <HTMLElement>document.getElementById('c-' + sId);
+    if (nextContent != null) {
+      nextContent.setAttribute('class', 'box step-content is-active');
+    }
+  }
+
+  addScheduledTask(formValues, nxtButton) {
+    /** request started */
+    this.ngProgress.start();
+    formValues['schedule_enabled'] = false,
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts]: I think default is false

File: src/app/components/core/scheduler/list-tasks/list-tasks.component.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { Component, OnInit } from '@angular/core';
 import { SchedulesService, AlertService } from '../../../../services/index';
 import { NgProgress } from 'ngx-progressbar';
+import { Router } from '../../../../../../node_modules/@angular/router';
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/list-tasks/list-tasks.component.ts]: node_modules ?

File: src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts (hunk 2 above)
Comment (monika-Dianomic) [reply src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts]: then what should be the value of `schedule_repeat`? (as it is must for `INTERVAL` type schedule)

Comment (praveen-garg) [reply src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts]: ask on UI, keep default and placeholder value 30

File: src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts [hunk 5]
```diff
@@ -195,9 +214,18 @@ export class AddTaskWizardComponent implements OnInit {
   addScheduledTask(formValues, nxtButton) {
     /** request started */
     this.ngProgress.start();
-    formValues['schedule_repeat'] = 30,
-    formValues['cmd_params'] = {};
-    this.schedulesService.createTask(formValues)
+    // total time with days and hh:mm:ss
+    const RepeatTime = this.taskForm.get('repeat_time').value !== ('' || undefined) ? Utils.convertTimeToSec(
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts]: repeatTime

File: src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts [hunk 6]
```diff
@@ -195,9 +214,18 @@ export class AddTaskWizardComponent implements OnInit {
...
+      this.taskForm.get('repeat_time').value, this.taskForm.get('repeat_day').value) : 0;
+    const payload = {
+      'name': formValues['name'],
+      'plugin': formValues['plugin'],
+      'type': formValues['type'],
+      'schedule_repeat': RepeatTime,
+      'schedule_type': formValues['schedule_type'],
+      'cmd_params': {}
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.ts]: cmd_params are not required any more., as FOGL-1560 is fixed. please verify

File: src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { AddTaskWizardComponent } from './add-task-wizard.component';
+
+describe('AddServiceWizardComponent', () => {
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/add-task-wizard/add-task-wizard.component.spec.ts]: AddTaskWizardComponent

File: src/app/services/schedules.service.ts [hunk 8]
```diff
@@ -138,4 +139,16 @@ export class SchedulesService {
...
       catchError((error: Response) => observableThrowError(error)));
   }
+
+  /**
+   * Create a task
+   *
+   * POST | /foglamp/scheduled/task
+   *
+   */
+  public createTask(payload: any) {
```
Comment (praveen-garg) [inline src/app/services/schedules.service.ts]: createScheduledTask will be better choice BTW!

File: src/app/components/core/scheduler/scheduler.module.ts [hunk 9]
```diff
@@ -38,7 +46,8 @@ const routes: Routes = [
     RouterModule.forChild(routes),
     NgProgressModule,
     PipesModule,
-    AlertDialogModule
+    AlertDialogModule,
+    SharedModule
```
Comment (praveen-garg) [inline src/app/components/core/scheduler/scheduler.module.ts]: SharedModule is now being used here? and for?

Comment (monika-Dianomic) [reply src/app/components/core/scheduler/scheduler.module.ts]: For `ViewConfigItemComponent`

**Pull request merged**